dvc/command/repro.py

```diff
@@ -182,6 +183,12 @@ def add_arguments(repro_parser):
         default=False,
         help="Allows targets containing shell-style wildcards.",
     )
+    repro_parser.add_argument(
+        "--no-shell",
+        action="store_true",
+        default=False,
+        help="Executes command without shell, directly.",
+    )
```

Comment:
This is not needed. Repro should use params recorded in dvcfile. `shell: true/false` should be added to dvcfile (written by `dvc run`) so repro could use it.

Reply:
@efiop, should not this be a runtime option, rather than a dvcfile thing.

Reply:
@skshetry No, it should be recorded into dvcfile and then used in `repro`. Otherwise, you pretty much break repro.

Reply:
So, if `dvc run` executed with `--no-shell` option, it should add line to dvcfile and `repro` should catch and use it. 
An I right?

Reply:
@Melevir Correct.

Reply:
Just a bit more context. You create a pipeline with `dvc run` and expect `dvc repro` to be able to reproduce it (the whole pipeline). So `dvc repro --no-shell` will affect all commands in your pipeline, which is clearly undesired, as that might break some commands. `shell: true/false` is a command-level option, so it should be recorded. Same as in things like GHA, where it is per-command option.

Reply:
Sounds legit. Thanks for explanation.

Reply:
Just to make it clear: we are adding `shell` line on top of file, at the same level as `stages` since it works for all commands. Am I right?
It looks like I should also update schema of the file.

Reply:
@Melevir Actually no, `shell/no-shell` is cmd-level, so it shouldn't be applied to all commands/stages. It should be an optional field that complements each `cmd`. E.g.
```
stages:
  features:
    cmd: myfeaturescmd
    ...
  training:
    cmd: mytrainingcmd
    shell: false
    deps: ....
```